File: play-2.4.x/K-Modules/kmodules/K-Base/app/K/Aop/annotations/JsonApi.java
package K.Aop.annotations;

import play.mvc.With;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by kk on 14-6-10.
 */
@With(K.Aop.actions.JsonApiAction.class)
@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
public @interface JsonApi {

    boolean Transactional() default true;

    boolean UseEtag() default false;

    Class ReplyClass();

    String ApiMethodType() default "GET";

    Class PostDataClass() default Object.class;
}

File: play-kotlin-2.5.x/PlayKotlin/kmodules/K-Base/app/K/Aop/annotations/JsonApi.kt
package K.Aop.annotations

import K.Controllers.ApiDoc.SampleForm.PostFormSample
import play.mvc.With
import kotlin.reflect.KClass

/**
 * Created by kk on 14-6-10.
 */
@With(K.Aop.actions.JsonApiAction::class)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class JsonApi(val Transactional: Boolean = true, val UseEtag: Boolean = false, val ReplyClass : KClass<*>, val ApiMethodType:

String = "GET", val PostDataClass : KClass<*> = PostFormSample::class)
